src/Aacotroneo/Saml2/Http/Controllers/Saml2Controller.php

```diff
@@ -218,31 +218,62 @@ private function processRequestMessage( $user, $redirectUrl )
                 $request = $this->createRequest( $message, $appUser );
                 $requestQueryParams = $this->getUrlParamString( $request );
             }
-            print_r($redirectUrl . $requestQueryParams);
-            exit();
             return redirect( $redirectUrl . $requestQueryParams );
         }
         catch(Exception $e)
         {
             logger()->error('Saml2 error_detail', ['error' => $e->getMessage()]);
             session()->flash('saml2_error_detail', [$e->getMessage()]);
-            return redirect(config('saml2_settings.errorRoute'));
+            return redirect( $this->getErrorRedirectionUrl(config('saml2_settings.errorRoute'), $e->getMessage()) );
+        }
+        catch( InvalidHL7SegmentException $e )
```

Comment:
Since the first catch is Exception, will this error ever called? Also too many copy/paste